strapi/strapi
# Expand nested components

### What does it do?

- Expands the limit of nesting components to 6 levels of nesting
- calculate the Depth of components and there child components

### Why is it needed?

- to have the possibility to put a number limit on nesting components

### How to test it?

- you should only be able to nest up to 6 components from the CTB
- you should not be able to add a component field to a component at the 6th level
<img width="866" alt="image" src="https://github.com/strapi/strapi/assets/37274596/7acaa178-f79a-4ff5-b625-1ad5b5bac5f2">

- you should only be shown components that you can add within the limit in the list of components

<img width="866" alt="image" src="https://github.com/strapi/strapi/assets/37274596/c9d6f481-3ec3-48fb-b29d-d5623742607d">
